client/utils/contentProfiles.ts

```diff
@@ -81,10 +82,52 @@ export function getUnusedProfileFields(
     profile: IEditorProfile,
     includeGroupCheck: boolean = true
 ): Array<IProfileFieldEntry> {
+    // Temporary FIXME: Will be removed once we add support for CVs as fields in coverage profiles
+    if (profile.name === 'coverage') {
+        const fieldsFromProfile = getProfileFields(profile);
+
+        return orderBy(
+            fieldsFromProfile
+                .filter((field) => (includeGroupCheck && field.field.group == null) || !field.field.enabled),
+            'name'
+        );
+    }
+
+    const customVocabularies = planningApi.vocabularies.getCustomVocabularies();
+    const vocabularyIds = customVocabularies.map((x) => x._id);
+    const vocabularyFields: Array<IProfileFieldEntry> = customVocabularies
+        .map((x, i) => ({
+            field: {
+                enabled: false,
+                group: undefined,
+                index: i,
+            },
+            name: x._id,
+            schema: {
+                type: 'custom_vocabulary',
+                required: false,
+            }
+        }));
+
+    const fieldsFromProfile = getProfileFields(profile);
+    const usedVocabularies = fieldsFromProfile.filter(
+        (fieldEntry) =>
+            fieldEntry.schema?.type === 'custom_vocabulary'
+            && vocabularyIds.includes(fieldEntry.name)
+            && fieldEntry.field.enabled
+    );
+    const unusedVocabularies = vocabularyFields.filter((field) =>
+        field.schema?.type === 'custom_vocabulary'
+        && !usedVocabularies.some((usedFieldEntry) => usedFieldEntry.name === field.name)
+    );
+
     return orderBy(
-        getProfileFields(profile).filter(
-            (item) => (includeGroupCheck && item.field.group == null) || !item.field.enabled
-        ),
+        fieldsFromProfile
+            .filter((field) => (
+                ((includeGroupCheck && field.field.group == null) || !field.field.enabled) &&
```

Comment:
use functions/variables/comments to make that filter more understandable to read